ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -845,6 +852,91 @@ namespace OrdinalAxis {
             return groupPositions;
         };
 
+        /**
+         * Get index of point inside the extended ordinal positions array.
+         *
+         * @private
+         * @function Highcharts.Axis#getIndexOfPoint
+         *
+         * @param {number} val
+         *        The pixel value of a point.
+         *
+         * @param {Array<number>} [ordinalPositions]
+         *        An array of points visible in the current plot area.
+         *
+         * @param {Array<number>} [extendedOrdinalPositions]
+         *        An array of all points available on the axis
+         *        for the given data set.
+         *
+         * @return {number}
+         */
+        axisProto.getIndexOfPoint = function (
+            val: number,
+            ordinalPositions: Array<number>,
+            extendedOrdinalPositions: Array<number>
+        ): number {
+            const axis = this,
+                // Distance in pixels between two points
+                // on the ordinal axis in the current zoom.
+                ordinalPointPixelInterval = axis.len / ordinalPositions.length - 1,
+                shiftIndex = val / ordinalPointPixelInterval;
+
+            // Make sure that the returned index does not exceed array
+            // boundaries. If so, return the first or the last index of array.
+            return Math.min(
+                Math.max(
+                    extendedOrdinalPositions.indexOf(axis.min as any) + shiftIndex,
```

Comment:
I think, that `extendedOrdinalPositions.indexOf(axis.min as any)` can return -1. E.g. if you set the extremes using navigator